### ISSUE TITLE
V2 Issue: useFetcher generic not working

### ISSUE THREAD

**Issue by kentcdodds** [remix-run/remix#7251]
### What version of Remix are you using?

0.0.0-nightly-49e8da1-20230823

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

[Here's the TS Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARFDvmQNwBQokscA3nAFaoQB2cAvnCIlyAAUohgADwC0UAK7cA9NwgATLLTiLFcGAE8wWVAC44ABgB0ls9O7AA5gAsYAGz3SALAE4sADlXIAIzSAExmIQDMZr6RDODQ8GxyqFgAYlgwuI5YUPyCxKRkYlgSMvJKlHgwmtq6Bkam1la2Ds5unj7+QaHhUTERdHRYkkzwaHrcuIIK+MA8KLM8ABQAlKx0cHCUMHJQvJzLbPaUGcDc9qYA5NkuLhCX-Cv0fIPDo3DqBMhyLoQzMHNeBgIHIYFgAMIkSDcLDcGCrdabXA8VCEDJZHJwAC8cGSaXR2SgAB59IYIAQFgCeAA+Vb0DZwZHcLguLAWO72JYEAk5CwBGDIAD8FmOWFO5yeg02212vCJqmAADdqai5AQCETFArlc86EA):

```tsx
import * as React from "react";
import { json } from "@remix-run/node"; // types: 0.0.0-nightly-49e8da1-20230823
import { useFetcher } from "@remix-run/react"; // types: 0.0.0-nightly-49e8da1-20230823

export async function action() {
  return json({ greeting: 'hello' });
}

export default function RouteComponent() {
  const fetcher = useFetcher<typeof action>();

  console.log(fetcher.data?.greeting); // <-- Property 'greeting' does not exist on type '() => Promise<TypedResponse<{ greeting: string; }>>'.(2339)


  return <div>stuff</div>;
}
```

### Expected Behavior

I should be able to get type inference from the `useFetcher` generic

### Actual Behavior

It's not able to infer properties.

**Comment by brophdawg11** [comment]
Thanks for the alpha testing!!

**Comment by brophdawg11** [comment]
Should be fixed by #7257 in tonights release

**Comment by kentcdodds** [comment]
Thanks a lot!
